pkg/fileutils/fileutils_darwin.go

```diff
@@ -25,3 +25,13 @@ func GetTotalUsedFds() int {
 
 	return len(fds) - 1
 }
+
+// PathExists checks whether a file or directory exists at the given path.
+func PathExists(path string, followSymlinks bool) error {
```

Comment:
Should the name of this function just return an err.

err:=fileutils.Exist(PATH);
err = nil -> True
err == ENOEXIST == False
err == (!nil && not ENOEXIST) == err.

Reply:
I didn't change the type of the returned value because it is easier to replace the existing calls to `os.Stat()`.  Every existing call to `_, err := os.Stat(...)` simply becomes `err := fileutils.PathExists(..., true)` and the existing code for handling `err` is the same.

Reply:
I don't see the difference, since my function will do the same, except there is two functions, Exists, and Lexists, which is similar to Stat and Lstat.

Reply:
pushed a new version with Exist and Lexist